src/lang/nodejs/src/index.test.js

```diff
@@ -1,192 +1,281 @@
 const blot = require('./index.js');
-const WebSocket = require('ws');
+const errors = require('./errorMessages.js');
+const WebSocketClient = require('ws');
 const fs = require('fs');
 let ws = null;
 
 beforeAll(() => {
   blot.setPort(3000).openManually();
-  ws = new WebSocket('ws://localhost:3000');
+  ws = new WebSocketClient('ws://localhost:3000');
 });
 
 afterAll(() => {
   ws.close();
 });
 
-describe('Non JSON', () => {
-
+describe('Falsy JSON', () => {
   test('undefined', () => {
-    expect(() => blot.visualise(undefined)).toThrow('Visualise must take in a valid JSON value')
+    expect(() => blot.visualise(undefined)).toThrow(errors.invalidJSONError);
   });
 
   test('Function', () => {
-    expect(() => blot.visualise(() => "hello world")).toThrow('Visualise must take in a valid JSON value');
+    expect(() => blot.visualise(() => 'hello world')).toThrow(
+      errors.invalidJSONError
+    );
+  });
+
+  test('Tests null', () => {
+    expect(() => blot.visualise(null)).toThrow(errors.invalidJSONError);
+  });
+
+  test('Empty String', () => {
+    expect(() => blot.visualise('')).toThrow(errors.invalidJSONError);
   });
 });
 
-describe('Standard JSON tests', () => {
+describe('Invalid JSON tests', () => {
+  test('Object', () => {
+    expect(() =>
+      blot.visualise({
+        name: 'John'
+      })
+    ).toThrow(errors.invalidJSONError);
+  });
+
+  test('Array', () => {
+    expect(() => blot.visualise([1, 2, 3, 4])).toThrow(errors.invalidJSONError);
+  });
+
+  test('Plain String', () => {
+    expect(() => blot.visualise('hello world')).toThrow(
+      errors.invalidJSONError
+    );
+  });
+});
 
-  test('Tests empty string', (done) => {
+describe('Standard JSON tests', () => {
+  test('Stringified empty string', (done) => {
+    const emptyString = '""';
```

Comment:
```suggestion
    const testData = '""';
```
I think testData would be a better name, to make the test more easily understandable. That the test data is an empty can be instantly determined.